src/sentry/metrics/datadog.py

```diff
@@ -50,9 +43,8 @@ def incr(self, key, instance=None, tags=None, amount=1, sample_rate=1):
             tags['instance'] = instance
         if tags:
             tags = ['{}:{}'.format(*i) for i in tags.items()]
-        # datadog does not implement sampling here
-        amount = _sampled_value(amount, sample_rate)
         self.stats.increment(self._get_key(key), amount,
+                             sample_rate=sample_rate,
```

Comment:
This technically doesn't do anything at all, but I'd rather us defer to the datadog library itself to implement this if they choose to. I assume they didn't since incrementing things in memory is cheap and doesn't really matter.

See: https://github.com/DataDog/datadogpy/blob/master/datadog/threadstats/metrics.py#L132